portal-bridge/src/types/state/content.rs

```diff
@@ -24,21 +27,123 @@ pub fn create_content_key(account_proof: &AccountProof) -> anyhow::Result<StateC
     }))
 }
 
-pub fn create_content_value(
+pub fn create_account_content_value(
     block_hash: B256,
-    account_proof: &AccountProof,
+    account_proof: &TrieProof,
 ) -> anyhow::Result<StateContentValue> {
-    let trie_proof = TrieProof::from(
-        account_proof
-            .proof
-            .iter()
-            .map(|bytes| EncodedTrieNode::from(bytes.to_vec()))
-            .collect::<Vec<EncodedTrieNode>>(),
-    );
+    let trie_proof = EncodedTrieProof::from(account_proof);
```

Comment:
I think you should remove this line and simplify this ever further:
```rust
Ok(StateContentValue::AccountTrieNodeWithProof(
    AccountTrieNodeWithProof {
        proof: account_proof.into(),
        block_hash,
    },
))
```

Also in several places below.